src/algebra/category/Module/change_of_rings.lean

```diff
@@ -161,4 +165,92 @@ variables {R : Type u₁} {S : Type u₂} [comm_ring R] [comm_ring S] (f : R →
 
 end extend_scalars
 
+namespace coextend_scalars
+
+variables {R : Type u₁} {S : Type u₂} [ring R] [ring S] (f : R →+* S)
+
+section unbundled
+
+variables (M : Type v) [add_comm_monoid M] [module R M]
+
+local notation `Hom` M := (restrict_scalars f).obj ⟨S⟩ →ₗ[R] M
```

Comment:
I think I would prefer if `R` and `S` were visible in the notation. It might also work to only introduce notation for
```
(restrict_scalars f).obj ⟨S⟩
```
and then simply write
```
foo →ₗ[R] M
```
where `foo` is the newly introduced notation.

Maybe something like
```
-- We use `S'` to denote `S` viewed as `R`-module, via the map `f`.
local notation `S'` := (restrict_scalars f).obj ⟨S⟩
```
would be enough?